src/deepsparse/analyze.py

```diff
@@ -0,0 +1,328 @@
+        type=str,
+    )
+    parser.add_argument(
+        "--optimization",
+        help="To enable or disable optimizations (Tensor Columns)",
+        type=bool,
+        default=True,
+    )
+    parser.add_argument(
+        "-i",
+        "--input_shapes",
+        help="Override the shapes of the inputs, "
+        'i.e. -shapes "[1,2,3],[4,5,6],[7,8,9]" results in '
+        "input0=[1,2,3] input1=[4,5,6] input2=[7,8,9]",
+        type=str,
+        default="",
+    )
+    parser.add_argument(
+        "-q",
+        "--quiet",
+        help="Lower logging verbosity",
+        action="store_true",
+        default=False,
+    )
+    parser.add_argument(
+        "-x",
+        "--export_path",
+        help="Store results into a JSON file",
+        type=str,
+        default=None,
+    )
+
+    return parser.parse_args()
+
+
+def layer_info_to_string(li, format_str):
+    if li["name"] == "sub_pyramid":
+        return format_str.format(li["name"], "[]", "[]", "[]", 0, 0, 0, 0, "")
+    else:
+        return format_str.format(
+            li["name"],
+            "{}".format(list(li["output_dims"].values())),
+            "{}".format(list(li["kernel_dims"].values())),
+            "{}".format(list(li["strides"].values())),
+            li["activation_sparsity"],
+            li["average_run_time_in_ms"],
+            li["average_utilization"] * 100.0,
+            li["average_teraflops_per_second"],
+            li["canonical_name"],
+        )
+
+
+def construct_layer_table(result):
+    table_str = (
+        "Name                        | OutDims                    | "
+        "KerDims                    | Strides      | ActSpars | "
+        "Time(ms) |  Util(%) | TFLOPS   | Canonical Name\n"
+    )
+    info_format_base = (
+        "{:26} | {:26} | {:12} | {: >#08.4f} | "
+        "{: >#08.4f} | {: >#08.4f} | {: >#08.4f} | {:12}"
+    )
+    for li in result["layer_info"]:
+        table_str += layer_info_to_string(
+            li,
+            "{:28}| " + info_format_base + "\n",
+        )
+        for sub_li in li["sub_layer_info"]:
+            table_str += layer_info_to_string(
+                sub_li,
+                "  {:26}| " + info_format_base + "\n",
+            )
+
+    table_str += "Total Time(MS): {:05f}\n".format(result["average_total_time"])
+    table_str += "Items per second: {:05f}\n".format(result["items_per_second"])
+    table_str += "Batch Size: {}\n".format(result["batch_size"])
+    table_str += "Number of threads: {}\n".format(result["num_threads"])
+
+    return table_str
+
+
+def process_line_item(total_layer_time, detailed_layer_time, li, strip_name):
+    if "average_run_time_in_ms" not in li:
+        # nothing to process
+        return
+
+    layer_type = li["name"]
+    if strip_name:
+        # peel off unique number
+        layer_type = layer_type.rsplit("_", 1)[0]
+        # peel off ks percentage
+        layer_type = layer_type.rsplit("-", 1)[0]
```

Comment:
I don't have a better suggestion but is this true for all units?   Is anything going to come out weird if there's a stray _ or something?

Reply:
this is true for all right now, subject to change obviously and this would at least do nothing if the special characters aren't present